lib/puppet/reports/http.rb

```diff
@@ -14,7 +14,7 @@
   def process
     url = URI.parse(Puppet[:reporturl])
     headers = { "Content-Type" => "application/x-yaml" }
-    options = {}
+    options = { :metric_id => [:puppet, :report, :http] }
```

Comment:
Yeah, it's fine that the `:metric-id` won't be used by the `http_pool` connection in Ruby Puppet code.  As long as its presence in the `get` or `post` options doesn't cause any problems, should be fine.  Where this will actually be used is in Puppet Server's http connection class - an alternate implementation registered only when running under Puppet Server.  See [this code](https://github.com/puppetlabs/puppetserver/blob/bb21b3824532d4c52295c348e8caf231ee584d76/src/ruby/puppetserver-lib/puppet/server/http_client.rb#L98-L100) for reference.